### ISSUE TITLE
ERROR: no such image: misolims/miso-lims-migration:: invalid reference format

### ISSUE THREAD

**Issue by ArnieBerg** [miso-lims/miso-lims#2402]
I get this error on both the plain and detailed samples.

$~/miso-lims-compose$ sudo docker-compose -f demo.plain.yml up
WARNING: The MISO_DB variable is not set. Defaulting to a blank string.
WARNING: The MISO_DB_USER variable is not set. Defaulting to a blank string.
WARNING: The MISO_TAG variable is not set. Defaulting to a blank string.
WARNING: The MISO_DB_PASSWORD_FILE variable is not set. Defaulting to a blank string.
WARNING: Service "db" uses an undefined secret file "/home/analynano/miso-lims-compose", the following file sh
ould be created "/home/analynano/miso-lims-compose"
WARNING: Service "flyway" uses an undefined secret file "/home/analynano/miso-lims-compose", the following file should be created "/home/analynano/miso-lims-compose"
WARNING: Service "webapp" uses an undefined secret file "/home/analynano/miso-lims-compose", the following file should be created "/home/analynano/miso-lims-compose"
Creating network "miso-lims-compose_default" with the default driver
Pulling db (mysql:5.7.25)...
5.7.25: Pulling from library/mysql
27833a3ba0a5: Pull complete
864c283b3c4b: Pull complete
cea281b2278b: Pull complete
8f856c14f5af: Pull complete
9c4f38c23b6f: Pull complete
1b810e1751b3: Pull complete
5479aaef3d30: Pull complete
1d924ec3d520: Pull complete
1ab7ae63ac60: Pull complete
08aa5f3680e9: Pull complete
a832d0a0972a: Pull complete
Digest: sha256:dba5fed182e64064b688ccd22b2f9cad4ee88608c82f8cff21e17bab8da72b81
Status: Downloaded newer image for mysql:5.7.25
ERROR: no such image: misolims/miso-lims-migration:: invalid reference format

I installed the docker-compose version on Ubuntu 18.04.  Any suggestions?

**Comment by djcooke** [comment]
Your issue is the missing environment variables. See the warnings at the top of your log. Full instructions for launching MISO via Docker are here: https://miso-lims.readthedocs.io/projects/docs/en/latest/admin/compose-installation-guide/

**Comment by ArnieBerg** [comment]
Thanks for your quick response.

I should backtrack here.  You are correct in your response, but you will notice that I ran the docker-compose using 'sudo'.  I did that to try to avoid an earlier error, which I now list below.  I believe this should work, but it is reporting a permission error.

~/miso-lims-compose$ docker-compose -f demo.plain.yml up
Traceback (most recent call last):
  File "urllib3/connectionpool.py", line 677, in urlopen
  File "urllib3/connectionpool.py", line 392, in _make_request
  File "http/client.py", line 1252, in request
  File "http/client.py", line 1298, in _send_request
  File "http/client.py", line 1247, in endheaders
  File "http/client.py", line 1026, in _send_output
  File "http/client.py", line 966, in send
  File "docker/transport/unixconn.py", line 43, in connect
PermissionError: [Errno 13] Permission denied

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "requests/adapters.py", line 449, in send
  File "urllib3/connectionpool.py", line 727, in urlopen
  File "urllib3/util/retry.py", line 403, in increment
  File "urllib3/packages/six.py", line 734, in reraise
  File "urllib3/connectionpool.py", line 677, in urlopen
  File "urllib3/connectionpool.py", line 392, in _make_request
  File "http/client.py", line 1252, in request
  File "http/client.py", line 1298, in _send_request
  File "http/client.py", line 1247, in endheaders
  File "http/client.py", line 1026, in _send_output
  File "http/client.py", line 966, in send
  File "docker/transport/unixconn.py", line 43, in connect
urllib3.exceptions.ProtocolError: ('Connection aborted.', PermissionError(13, 'Permission denied'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "docker/api/client.py", line 205, in _retrieve_server_version
  File "docker/api/daemon.py", line 181, in version
  File "docker/utils/decorators.py", line 46, in inner
  File "docker/api/client.py", line 228, in _get
  File "requests/sessions.py", line 543, in get
  File "requests/sessions.py", line 530, in request
  File "requests/sessions.py", line 643, in send
  File "requests/adapters.py", line 498, in send
requests.exceptions.ConnectionError: ('Connection aborted.', PermissionError(13, 'Permission denied'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "bin/docker-compose", line 3, in <module>
  File "compose/cli/main.py", line 67, in main
  File "compose/cli/main.py", line 123, in perform_command
  File "compose/cli/command.py", line 69, in project_from_options
  File "compose/cli/command.py", line 132, in get_project
  File "compose/cli/docker_client.py", line 43, in get_client
  File "compose/cli/docker_client.py", line 170, in docker_client
  File "docker/api/client.py", line 188, in __init__
  File "docker/api/client.py", line 213, in _retrieve_server_version
docker.errors.DockerException: Error while fetching server API version: ('Connection aborted.', PermissionError(13, 'Permission denied'))
[11328] Failed to execute script docker-compose
